firmware/library/L2_HAL/memory/sddriver.hpp

```diff
@@ -0,0 +1,1169 @@
+
+      // Check each bit
+      for (uint16_t j = 0; j < 8; j++)
+      {
+        // Check if we will need to add the polynomial with no carry
+        // (i.e. XOR it)
+        bool xorIsNeeded = crc & (1 << 15);
+
+        // Left-shift the crc
+        crc = crc << 1;
+
+        // Check if msb is 1
+        if (xorIsNeeded)
+        {
+          // XOR it with the polynomial
+          crc ^= table.kPoly16bit;
+        }
+      }
+
+      // Add crc to the table
+      table.crc_table[i] = crc;
+    }
+
+    // Return the CRC-16 table
+    return table;
+  }
+
+  // @description     sends a command frame to the SD Card and sets the
+  //                  response
+  // @parameter       (Command) sdc                 the command code to
+  //                                                  send as the first byte
+  // @parameter       (uint32_t) arg                  the 4-byte argument
+  //                                                  for the command
+  // @parameter       ~(uint8_t*) response_buffer     pointer to a buffer to
+  //                                                  house the response
+  //                                                  frame
+  // @parameter       (uint32_t) delay                delay (ms) between
+  //                                                  asserting CS and
+  //                                                  sending a cmd frame
+  // @parameter       (KeepAlive) keep_alive          determines whether to
+  //                                                  de-assert CS after
+  //                                                  receiving response
+  // @returns         on success: the number of bytes given in the response
+  //                  on failure: -1
+  virtual uint32_t SendCmd(Command sdc, uint32_t arg, uint8_t response_buffer[],
+                           uint32_t delay, KeepAlive keep_alive) = 0;
+
+  // @description     initializes all required hardware constructs to talk
+  //                  to the SD Card via SPI/SSP
+  virtual void Initialize() = 0;
+
+  // @description     runs the SD Card initialization sequence, effectively
+  //                  resetting communication with the card and mounting it
+  //                  to the board
+  // @parameter       (CardInfo_t*) sd                  a pointer to the SD
+  //                                                  Card data structure
+  // @returns         true on init success
+  virtual bool Mount(CardInfo_t * sd) = 0;
+
+  // @description     reads any number of 512-byte blocks from the SD Card
+  //                  and returns the received block
+  //                  This function assumes the block length is locked to
+  //                  512-bytes (as is the case for SDHC/SDXC cards).
+  //                  SD Command 17 and 18 are are used for single and multi
+  //                  block reads, respectively.
+  // @parameter       (uint32_t) address      The address of the starting
```

Comment:
```suggestion
  // @parameter       address      The address of the starting
```
Remove all types from parameters from all of the documentation comments in the code.